src/lib/reasoners/intervals.mli

```diff
@@ -19,140 +19,137 @@
 (*                                                                        *)
 (*     CNRS - INRIA - Universite Paris Sud                                *)
 (*                                                                        *)
-(*     Until 2013, some parts of this code were released under            *)
-(*     the Apache Software License version 2.0.                           *)
-(*                                                                        *)
 (*     ---------------------------------------------------------------    *)
 (*                                                                        *)
 (*     More details can be found in the directory licenses/               *)
 (*                                                                        *)
 (**************************************************************************)
 
-type t
-
-exception NotConsistent of Explanation.t
-exception No_finite_bound
+open Intervals_intf
 
-val undefined : Ty.t -> t
+(** This module provides implementations of union-of-intervals over reals and
+    integers. *)
 
-val is_undefined : t -> bool
+type 'a union
+(** Polymorphic union type. This allows writing conversion functions between
+    integer and real unions. *)
 
-val point : Numbers.Q.t -> Ty.t -> Explanation.t -> t
+module Real : AlgebraicField
```

Comment:
My heart of arithmetician cries when I see this.

Reply:
I am not happy with this name either but couldn't find anything better, even if not strictly correct mathematically (the only other good alternative I had was `FieldWithNthRoot`, which is a mouthful). I happily take suggestions, maybe `RadicalField`?

Reply:
`RadicalField` sounds a little bit better because there exists a notion of `Radical extension` in `Galois Theory` and our `Real` field is actually a tower of radical extensions of the rational field.

Let's keep the old name. As we have a comment to explain this field is closed by radical extraction, it's acceptable.